src/ui/Folding/Folding.ts

```diff
@@ -334,29 +352,45 @@ export class Folding extends Component {
       if (resultNodes[i].result.uniqueId == uniqueId) {
         return resultNodes[i];
       }
-      let resultNode = Folding.findUniqueId(resultNodes[i].children, uniqueId);
+      const resultNode = Folding.findUniqueId(resultNodes[i].children, uniqueId);
       if (resultNode != null) {
         return resultNode;
       }
     }
     return null;
   }
 
+  private switcherooFoldingFields() {
+    // Swap "old" childField and parentField and assign them to the "new" parent option
+    // This needs to be done because connectors push the default data in *reverse* order compared to what the index expect.
+    if (this.options.childField != null) {
+      this.logger.warn('Detecting usage of deprecated option "childField". Assigning it automatically to the "parent" option instead.');
+      this.logger.warn('To remove this warning, rename the "childField" option (data-child-field) to "parent" (data-parent)');
+      this.options.parent = this.options.childField;
+    }
+
+    if (this.options.parentField != null) {
+      this.logger.warn('Detecting usage of deprecated option "parentField". Assigning it automatically to the "child" option instead.');
+      this.logger.warn('To remove this warning, rename the "parentField" option (data-parent-field) to "child" (data-child)');
+      this.options.child = this.options.parentField;
+    }
+  }
+
   private handleBuildingQuery(data: IBuildingQueryEventArgs) {
     Assert.exists(data);
 
     if (!this.disabled) {
-      data.queryBuilder.childField = <string>this.options.childField;
-      data.queryBuilder.parentField = <string>this.options.parentField;
+      data.queryBuilder.childField = <string>this.options.parent;
```

Comment:
Just to make sure, you did not update the queryBuilder fields `childField` and `parentField` here because that would involve wider changes, I guess even in the search api side?

Reply:
Exactly, the errors comes from the connectors side ; We're simply hiding that mistake in the UI because it's simpler than re-indexing.

So, the actual good name for the fields as far as the search API and index is concerned is the "old" childField and parentField.